File: leetcode/71-简化路径/71-简化路径.js
/**
 * @param {string} path
 * @return {string}
 */

// 71-简化路径
var simplifyPath = function (path) {
    const paths = path.split("/");
    const stack = [];
    for (let i = 0; i < paths.length; i++) {
        if (paths[i] === "..") {
            if (stack.length > 0) {
                stack.pop();
            }
        } else if (paths[i] != "" && paths[i] != ".") {
            stack.push(paths[i]);
        }
    }
    return "/" + stack.join("/");
};

File: leetcode/71-简化路径/71-简化路径.py
# 71-简化路径

class Solution:
    def simplifyPath(self, path: str) -> str:
        paths = path.split("/")
        stack = []
        for i in paths:
            if i == "..":
                if len(stack) > 0:
                    stack = stack[:len(stack)-1]
            elif i !=""  and i != ".":
                stack.append(i)

        return "/" + "/".join(stack)
             

        